webui.py

```diff
@@ -28,7 +28,16 @@
 
 startup_timer.record("import torch")
 
+import requests
+def gradio_get(url, **kwargs):
+    kwargs.setdefault('allow_redirects', True)
+    return requests.api.request('get', 'http://127.0.0.1/', **kwargs)
+
+original_get = requests.get
+requests.get = gradio_get
 import gradio
+requests.get = original_get

```

Comment:
As an aside, this could probably be implemented as something like

```suggestion
import requests
from unittest.mock import patch
with patch("requests.request", side_effect=requests.ConnectionError()):
    import gradio
```